src/Simplify_Call.cpp

```diff
@@ -20,6 +20,40 @@ Expr Simplify::visit(const Call *op, ExprInfo *bounds) {
         } else {
             return strict_float(arg);
         }
+    } else if (op->is_intrinsic(Call::popcount) ||
+               op->is_intrinsic(Call::count_leading_zeros) ||
+               op->is_intrinsic(Call::count_trailing_zeros)) {
+        Expr a = mutate(op->args[0], nullptr);
+
+        uint64_t ua = 0;
+        if (const_int(a, (int64_t *)(&ua)) || const_uint(a, &ua)) {
+            const int bits = op->type.bits();
+            // __builtin_clz and friends only operate on uint;
```

Comment:
__builtin_clzll ?

Reply:
Heh, didn't occur to me to check for longer variants. I'll sub those in and verify that they work on MSVC.

Reply:
Note that in addition to using the intrinsics, this now checks for static clz(0)/ctz(0) in the simplifier and makes it a compile error, and only does the new clz optimization in Bounds if we can prove the value is nonzero.